## Use Case: Finding the Angle Between Two Vectors
You can use np.arccos() to compute the angle between two vectors using the dot product formula:
```math 
\theta = \cos^{-1} \left( \frac{a \cdot b}{\lvert{a}\rvert \cdot \lvert{b}\rvert} \right)

In [6]:
# Compute the Angle Between Two 2D Vectors

import numpy as np

# Define two vectors
a = np.array([1, 0])
b = np.array([0, 1])

# Compute dot product
dot_product = np.dot(a, b)

# Compute magnitudes (norms)
norm_a = np.linalg.norm(a)
norm_b = np.linalg.norm(b)

# Compute angle in radians
theta = np.arccos(dot_product / (norm_a * norm_b))

# Convert to degrees
theta_degrees = np.degrees(theta)

print("Angle in radians:", theta)
print("Angle in degrees:", theta_degrees)


Angle in radians: 1.5707963267948966
Angle in degrees: 90.0


## Radians Explanation

#### 📌 What Are Radians?

A radian is a unit of angular measure based on the radius of a circle. It represents the angle subtended at the center of a circle by an arc ***equal in length to the radius*** of the circle.

[See this clip to see 1 radian, and more about radiant](https://youtu.be/TGC4K6AMf7Y?si=W_eCwHnWLF0nYjNQ&t=107)

Mathematically, 
```math
\theta_{radian} = S/r \text{  ( where S is the arc that subtends the angle, and r is radius of that circle ) }
```

🔹 Key Facts About Radians:

- One full circle = 2𝜋 radians (≈ 6.283 radians).
- Half a circle (180°) = 𝜋 radians (≈ 3.1416 radians).
- Quarter circle (90°) = 𝜋/2 radians (≈ 1.5708 radians).
- 1 radian ≈ 57.2958 degrees (since 180∘ = 𝜋 radians).

🔥 Radians vs Degrees:

| Degrees (°) | Radians |
|------------|------------|
| 0°  | 0 rad |
| 30° | 𝜋/6 rad |
| 45° | 𝜋/4 rad |
| 60° | 𝜋/3 rad |
| 90° | 𝜋/2 rad |
| 180° | 𝜋 rad |
| 270° | 3𝜋/2 rad |
| 360° | 2𝜋 rad |


✅ Converting Between Degrees and Radians in Python

Use numpy.radians() and numpy.degrees() for conversion:

```python
import numpy as np

# Convert degrees to radians
radians = np.radians(180)
print(radians)  # Output: 3.141592653589793 (π)

# Convert radians to degrees
degrees = np.degrees(np.pi / 2)
print(degrees)  # Output: 90.0
```

## Why Use Radians?

Radians are more natural in mathematics and physics, especially in trigonometry and calculus, where functions like sin, cos, and tan work best in radian mode. 🚀



In [5]:
np.degrees(1.5707963267948966)

np.float64(90.0)